src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/PackageBuilder.cs

```diff
@@ -710,7 +710,7 @@ private static void ExcludeFiles(List<PhysicalPackageFile> searchFiles, string b
 
         private static void CreatePart(ZipArchive package, string path, Stream sourceStream)
         {
-            if (PackageHelper.IsNuspec(path) || ProjectJsonPathUtilities.IsProjectConfig(path))
```

Comment:
Yes, excluding the .nuspec makes total sense. That special case is important. I guess I am wondering why we had the project.json special case to begin with.